console_conf/controllers/identity.py

```diff
@@ -106,7 +106,12 @@ def write_login_details(fp, username, ips):
     sshcommands = "\n"
     for ip in ips:
         sshcommands += "    ssh %s@%s\n" % (username, ip)
-    tty_name = os.ttyname(0)[5:]  # strip off the /dev/
+    # handle error if we cannot get tty, e.g. in snap confinement
+    # try to get correct tty from env, it should be ready there
+    try:
+        tty_name = os.ttyname(0)[5:]  # strip off the /dev/
+    except OSError:
+        tty_name = os.getenv("tty", "")
```

Comment:
1. Are we sure it's lower case?  If it's upper case, this will not obtain the correct value.
2. if we don't get a tty_name, we're going to print something like
```Ubuntu Core 24 on 1.2.3.4 ()```
3. While this sequence is short, it's also duplicated, so I'd like to see it extracted out to a utility function.

Reply:
env variable is set by the supporting script which is invoked by the console-conf wrapper, so we can use here what we prefer
Original script is setting lower case so I maintained that

Reply:
if tty is set, is that the value intended to be used in preference to what os.ttyname() reports?

Reply:
Good question. TBH I have no strong opinion here.
Favouring `os.ttyname` means preserving current behaviour when possible, which was my goal, to have the least obstructive change and only introduce fallback.

But now when you ask, I suppose setting `tty` env would provide the possibility to override default behaviour, which might be sometimes useful.
Happy to change the logic if you prefer it that way.